### PR TITLE
Frontend search enhancements

### PR DESCRIPTION
**Pull request by richard67**
Pull Request for Issue #20 .

### Summary of Changes

This PR is to collect, discuss and text the enhancements of the frontend search as defined in isssue #20 .

Requirements:
- [x] with label and / or icon
- [x] in extended form 
- [x] as a component
- [x] With search results 
- [x] on different places 

### Testing Instructions

Checkout the branch, or apply the changes of this branch to a current 4.0-dev branch or recent nightly of the CMS.

You can do that with patchtester 4:

patchtester-cassiopeia

(to be completed)

### Expected result

(to be completed)

### Actual result

(to be completed)

### Documentation Changes Required

To be defined.

### CODE REVIEW

**Comment by richard67** [comment]
Changed back to draft so we don't accidently merge it.

File: components/com_finder/tmpl/search/default_form.php [hunk 1]
```diff
@@ -27,6 +27,9 @@
 <form action="<?php echo Route::_($this->query->toUri()); ?>" method="get" class="js-finder-searchform">
 	<?php echo $this->getFields(); ?>
 	<fieldset class="com-finder__search word mb-3">
+    <legend class="com-finder__search-legend sr-only">
+	    <?php echo Text::_('COM_FINDER_SEARCH_FORM_LEGEND'); ?>
+    </legend>
```
Comment (richard67) [inline components/com_finder/tmpl/search/default_form.php]:
```suggestion
	<legend class="com-finder__search-legend sr-only">
		<?php echo Text::_('COM_FINDER_SEARCH_FORM_LEGEND'); ?>
	</legend>
```
Please use tabs for indentation.

File: components/com_finder/tmpl/search/default_form.php [hunk 2]
```diff
@@ -49,7 +52,10 @@
 	</fieldset>
 
 	<?php if ($this->params->get('show_advanced', 1)) : ?>
-		<div id="advancedSearch" class="com-finder__advanced js-finder-advanced collapse<?php if ($this->params->get('expand_advanced', 0)) echo ' show'; ?>">
+		<fieldset id="advancedSearch" class="com-finder__advanced js-finder-advanced collapse<?php if ($this->params->get('expand_advanced', 0)) echo ' show'; ?>">
+      <legend class="com-finder__search-advanced sr-only">
+		    <?php echo Text::_('COM_FINDER_SEARCH_ADVANCED_LEGEND'); ?>
+      </legend>
```
Comment (richard67) [inline components/com_finder/tmpl/search/default_form.php]:
```suggestion
		<legend class="com-finder__search-advanced sr-only">
			<?php echo Text::_('COM_FINDER_SEARCH_ADVANCED_LEGEND'); ?>
		</legend>
```

**Comment by hans2103** [comment]
DL => OL conversion
DL was implemented in https://github.com/joomla/joomla-cms/pull/20572 by @Hackwar 
> I used a DL because I considered that the syntactically correct structure. There is no deeper reasoning.

Changing it from DL back to OL makes it easier for frontend developers to style each search result. 
Styling DT+DD is not as easy as styling a LI

**Comment by hans2103** [comment]
top right in Github: _At least 6 approving reviews are required to merge this pull request._

**Comment by richard67** [comment]
> top right in Github: _At least 6 approving reviews are required to merge this pull request._

That's not really relevant now since this PR here is just for development and maintenance but will not really be merged into the 4.0-dev branch.

**Comment by richard67** [comment]
I think this should be made as separate PR for the CMS, because it doesn't depend on or interact with our other developments here and touch the admin area. @hans2103 As the code originally comes from you: Can you make such PR for the 4.0-dev branch of the CMS?

**Comment by chmst** [comment]
Please ....  And ask the JAT team for a test.

**Comment by brianteeman** [comment]
There are quite a few changes in this PR that are not correct and definitely nothing to do with the template. Please lets not repeat the mistakes of the past and leave non-template changes to the main repo

**Comment by hans2103** [comment]
> Please .... And ask the JAT team for a test.

@chmst how to trigger them? Besides ask them via Glip.

> There are quite a few changes in this PR that are not correct ...

@brianteeman Can you be more specific. I will fix them. 

> ... and definitely nothing to do with the template. 

@brianteeman Planning to create multiple small PR's for joomla-cms repo

**Comment by brianteeman** [comment]
> @brianteeman Can you be more specific. I will fix them.

When the PR is in the correct place I will

**Comment by hans2103** [comment]
created PR https://github.com/joomla/joomla-cms/pull/30528 to solve the `legend` issue.

**Comment by hans2103** [comment]
created PR https://github.com/joomla/joomla-cms/pull/30534 to solve styling of search results

**Comment by hans2103** [comment]
created PR https://github.com/joomla/joomla-cms/pull/30535 to add statement of search results

**Comment by hans2103** [comment]
All changes of this PR have been converted to three separate Joomla/Joomla-CMS pull requests. 
Closing this PR

**Comment by brianteeman** [comment]
Thank you

**Pull request closed without merging**